### PR TITLE
fix: handle async loading of patch

### PR DESCRIPTION
**Pull request by neoziro**
Implement #738

### CODE REVIEW

**Comment by theKashey** [comment]
Hey! I've simplified babel. Look like it works well.

File: packages/react-hot-loader/src/patch.prod.js [hunk 1]
```diff
@@ -1 +1,4 @@
 /* noop */
+module.exports = {
+  default: false,
```
Comment (neoziro) [inline packages/react-hot-loader/src/patch.prod.js]: `null`

File: .eslintrc.js [hunk 2]
```diff
@@ -4,7 +4,13 @@ module.exports = {
   rules: {
     'no-underscore-dangle': [
       'error',
-      { allow: ['__REACT_HOT_LOADER__', '__standin_getCurrent'] },
+      {
+        allow: [
+          '__REACT_HOT_LOADER__',
+          '__standin_getCurrent',
+          '__registeredComponents',
```
Comment (neoziro) [inline .eslintrc.js]: Not needed.

File: packages/react-hot-loader/src/babel.js [hunk 3]
```diff
@@ -31,10 +31,11 @@ module.exports = function plugin(args) {
   const buildTagger = template(
     `
   var UNUSED = (function () {
-    if (typeof __REACT_HOT_LOADER__ === 'undefined') {
-      return;
+    var __REACT_HOT_LOADER__ = global.__REACT_HOT_LOADER__ || require('react-hot-loader/patch').default;
```
Comment (neoziro) [inline packages/react-hot-loader/src/babel.js]: Global does not work in browser.

**Pull request merged**